test/fail_compilation/test15703.d

```diff
@@ -0,0 +1,32 @@
+/*
+REQUIRED_ARGS: -m32
```

Comment:
Because I wanted the integral type to match in size the pointer type. But `size_t` comes out as `uint` or `ulong` depending on -m32 or -m64, and then the error messages don't match.